src/pages/ContributorFields.tsx

```diff
@@ -116,14 +142,32 @@ const ContributorFields: FC<ContributorFieldsProps> = ({ setAllChangesSaved }) =
                         <StyledTextField
                           {...field}
                           variant="filled"
+                          select
+                          required
                           label={t('type')}
+                          value={field.value}
                           error={touched && !!error}
                           helperText={<ErrorMessage name={field.name} />}
                           onBlur={(event) => {
                             handleBlur(event);
-                            !error && saveContributorField(event, contributor.identifier, index);
+                            setFieldTouched(
+                              `resource.contributors[${index}].features.dlr_contributor_type`,
+                              true,
+                              true
+                            );
                           }}
-                        />
+                          onChange={(event) => {
+                            handleChange(event);
+                            saveContributorField(event, contributor.identifier, index);
+                          }}>
+                          {contributorTypesTranslated.map((contributorType, index) => {
+                            return (
+                              <MenuItem key={index} value={contributorType.key}>
+                                <Typography variant="inherit">{contributorType.description}</Typography>
```

Comment:
tror du kan slippe unna fra å kjøre en generateContributorTypesTranslated() - ved å bruke `resource.contributor_type.${contributorType}` her isteden.

Reply:
Men går det an å sortere lista alfabetisk etter brukerens språkvalg da?

Reply:
ah. Men er det et krav ? Og blir listen oppdatert nå om brukeren endrer språk ?

Reply:
Er ikke krav, men er vel greit å ha? 
Kan man endre språk på sida uten at sida refreshes?

Reply:
det spørs hvordan vi implementerer språkskifte (er det i headeren blir det jo det) 
Men om du vil ha sortering, kan vi alltids legge på en useeffekt som lytter på språk-endringer

Reply:
Kommer til å bittelitt clunky, siden man faktisk trenger hele tabellen klar på mount. Hadde sorteringa i en useEffect som fulgte med på språk i første forsøk, men det som skjedde var at react begynte å klage på at default-verdien ikke eksisterte i dropdown menyen.
Men det er ikke et problem for min del: generere tabell i initialstate, for så å sortere den i useEffect er ikke noe stress.

Reply:
Sånn fikset. 
Fått også sjekket at det funger ved å ha en midlertidig useEffect inni ContributorFields kompnenten som satt å svitsjet språk hver gang man hentet frem kompnenten.